### PR TITLE
Current work-week shortcut & progress bar

### PR DESCRIPTION
**Pull request by waynew**
Not sure if anyone else is interested in this feature, but most of the time I'm interested in what I've done this current workweek (as opposed to the last week).

So I've added a shortcut to `report` that lets you do that with 3 extra keystrokes: ` -w`.

Many times I'm *also* interested if I've put in enough time this week. Sure I can look at the numbers and make a comparison, but a chart is easier to grok and since Click just had this one sitting around... why not? :trollface: 

This also adds a new option:

```
[options]
target_hours_per_week = 40  #unless you're a workaholic ;)
```

**With that option:**

```
⚘ watson report -w                    
Mon 09 November 2015 -> Fri 13 November 2015

watson - 45m 42s
        [personal         45m 42s]
        [progress bar     45m 42s]

Work completed: 0.8h/22.9h  [#-----------------------------------]    3%
```

**And without:**

```
⚘ watson report -w                   
Mon 09 November 2015 -> Fri 13 November 2015

watson - 45m 42s
        [personal         45m 42s]
        [progress bar     45m 42s]
```

---
## Possible Improvements

If this gets merged in, an idea for future enhancement would be the ability to have per-tag target hours in the config, e.g.

```
[options]
target_hours_per_week = 40

[tag:watson]
target_hours_per_week = 5
```

It could also be nice to provide alternate `workweek` config options for those with alternative work schedules, e.g.

```
[options]
workweek_start = Tuesday
workweek_end = Saturday
```

Or maybe even:

```
[options]
work_days = Monday, Wednesday, Friday
```

I don't really have a need for any of those options at the moment, but they might be useful to others.

### CODE REVIEW

**Comment by tomleo** [comment]
I really like this idea :+1:

File: watson/cli.py [hunk 1]
```diff
@@ -371,6 +381,23 @@ def report(watson, from_, to, projects, tags):
             style('time', '{}'.format(format_timedelta(total)))
         ))
 
+    target_hours_per_week = watson.config.getfloat('options',
+                                                   'target_hours_per_week')
+    if target_hours_per_week is not None:
+        sec_per_min = 60
+        min_per_hr = 60
+        offset = 1000
+        weeks = abs(span.start - span.stop).days/7
```
Comment (waynew) [inline watson/cli.py]: I just realized that this calculation isn't quite right. This works if there are 7 days in the workweek, but a typical week is 5 days.

**Comment by larsborn** [comment]
if no one opposes, I would take this one over. Or did it fell off the feature list entirely?

**Comment by SpotlightKid** [comment]
At this point I would suggest starting a new separate `watson-report` program and invest a little thought in how the command line interface can designed user-friendly and still with room for additions.

**Comment by larsborn** [comment]
so you suggest to not integrate such features into watson but write a separate program (that uses watson for example)?

**Comment by k4nar** [comment]
I don't know if we need an external command for that. Maybe a clever addons system could cover that.

**Comment by SpotlightKid** [comment]
The program could still live in the `watson` package. but I would add a separate module and command line entry point in `setup.py`.

**Comment by jmaupetit** [comment]
I think it's a good idea to decrease sub-commands complexity. I invite all of you to up/down vote previous @SpotlightKid comment to see where we are going on this.

**Comment by larsborn** [comment]
are there already more then this one "more advanced" report feature planed? 

What insistently springs to my mind is a report, that does the following for every "work day" (definition outstanding): subtract the numbers of hours worked from a configurable "target number" of hours. Then sum up these differences. This would enable easy overtime tracking. This is somehow related to this PR but also a bit different imho.

**Comment by SpotlightKid** [comment]
Other features I would like to see integrated / moved into a watson report command are:

* output in CSV or JSON format
* long form output with log messages (from pending PR #119)
* configurable rounding (#109)

**Comment by larsborn** [comment]
I feel, that we totally lost the "next action" in this thread. Since I am the new kid here, I'm totally happy with just following orders, so if someone tells me to put something into an external command, I'll do it ;-)

Sooo, what should we do?

**Comment by SpotlightKid** [comment]
* Read the [click documentation](http://click.pocoo.org/).
* Copy `watson/cliy.py` to `watson/report.py`.
* Throw out everything not related to the report command.
* Set up a setuptools console command entry point in `setup.py` for a command named `watson-report` or similar, which calls into this file.
* Design a command line API for the new command that integrates the existing functionality of `watson report` but gives a good foundation to add more report types and functions. Feel free to rename options and/or restructure arguments to achieve better consistency.
* Make a WIP PR for discussion.

**Comment by jmaupetit** [comment]
Thx for taking the lead on this @SpotlightKid !

**Pull request closed without merging**